sqlglot/optimizer/simplify.py

```diff
@@ -507,6 +509,9 @@ def simplify_literals(expression, root=True):
                 return exp.Literal.number(value[1:])
             return exp.Literal.number(f"-{value}")
 
+    if isinstance(expression, DATE_OPS):
```

Comment:
We can also get rid of `DATE_OPS` to avoid introducing more constants and instead do `type(expression) in INVERSE_DATE_OPS`

Reply:
Sure

Interesting test:

```python
class A:
    ...

class B:
    ...

class C:
    ...

class D:
    ...

SET = {A, B, C, D}
TYPES = (A, B, C, D)

b = B()

import timeit

setup = "from __main__ import SET, TYPES, b"
print(timeit.timeit("type(b) in TYPES", setup=setup, number=100000))
print(timeit.timeit("isinstance(b, TYPES)", setup=setup, number=100000))
```

Python 3.9:
```
type in   : 0.01205877
isinstance: 0.010168221999999998
```

Python 3.11:
```
type in   : 0.003846660954877734
isinstance: 0.004812692059203982
```

Reply:
Interesting indeed